Framework/PythonInterface/plugins/algorithms/CreatePoleFigureTableWorkspace.py

```diff
@@ -0,0 +1,231 @@
+            intensities = np.asarray(peak_param_ws.column("I"))
+            if not self.no_x0_needed:
+                x0s = np.asarray(peak_param_ws.column("X0"))
+            else:
+                x0s = np.ones((len(ab_arr)))
+            if not self.no_chi2_needed:
+                chi2 = np.asarray(peak_param_ws.column("chi2"))
+            else:
+                chi2 = np.zeros((len(ab_arr)))
+
+        else:
+            intensities = np.ones((len(ab_arr)))
+            chi2 = np.zeros((len(ab_arr)))
+            x0s = np.ones((len(ab_arr)))
+
+        if hkl_is_set:
+            hkl = _parse_hkl(hkl)
+            xtal = ws.sample().getCrystalStructure()
+            peak = xtal.getUnitCell().d(hkl)
+            scat_power = _calc_scattering_power(xtal, hkl)
+        else:
+            # if no hkl provided, peak will be set to mean x0 and scattering power will be 1
+            peak = np.mean(x0s)
+            scat_power = 1.0
+
+        # create the table to hold the data
+        table_ws = self.exec_child_alg("CreateEmptyTableWorkspace", OutputWorkspace="_tmp")
+        table_ws.addColumn(type="double", name="Alpha", plottype=2)
+        table_ws.addColumn(type="double", name="Beta", plottype=2)
+        table_ws.addColumn(type="double", name="Intensity", plottype=2)
+
+        # go through each detector group
+        for index, (a, b) in enumerate(ab_arr):
+            # check chi2 is under threshold (or threshold is set to be ignored) and peak position is also within tol
+            if ((chi2[index] < chi_thresh) or chi_thresh == 0.0) and ((np.abs(x0s[index] - peak) < x0_thresh) or x0_thresh == 0.0):
+                # amend intensity for any scattering correction
+                intensity = intensities[index] / scat_power
+                # add data to table
+                table_ws.addRow({"Alpha": a, "Beta": b, "Intensity": intensity})
+
+        self.setProperty("OutputWorkspace", table_ws)
+
+    def exec_child_alg(self, alg_name: str, **kwargs):
+        alg = self.createChildAlgorithm(alg_name, enableLogging=False)
+        alg.setAlwaysStoreInADS(False)
+        alg.initialize()
+        alg.setProperties(kwargs)
+        alg.execute()
+        out_props = tuple(alg.getProperty(prop).value for prop in alg.outputProperties())
+        return out_props[0] if len(out_props) == 1 else out_props
+
+
+def _parse_hkl(hkl: str) -> V3D:
+    return V3D(*[int("".join([char for char in part if char.isnumeric()])) for part in hkl.split(",")])
+
+
+def _calc_scattering_power(xtal, hkl) -> float:
+    generator = ReflectionGenerator(xtal)
+    d = xtal.getUnitCell().d(hkl)
+    f_sq = generator.getFsSquared(
+        [
+            hkl,
+        ]
+    )[0]
+
+    pg = xtal.getSpaceGroup().getPointGroup()
+    m = len(pg.getEquivalents(hkl))  # multiplicity
+
+    vol = xtal.getUnitCell().volume() / len(xtal.getScatterers())  # volume per atom
+    return (m * f_sq * d**4) / vol**2  # Eq 2 of NyRTeX paper
```

Comment:
Thanks for adding the Eq number, would it be possible to include the citation for the NyRTeX paper in the code somewhere, perhaps a docstring of this function?